src/logic/small.lean

```diff
@@ -47,14 +47,17 @@ nonempty.some (classical.some_spec (@small.equiv_small α _))
 
 @[priority 100]
 instance small_self (α : Type v) : small.{v} α :=
-small.mk' (equiv.refl _)
+small.mk' $ equiv.refl α
+
+theorem small_lift (α : Type u) [small.{v} α] : small.{max v w} α :=
+small.mk' $ (equiv_shrink α).trans equiv.ulift.symm
```

Comment:
`small_ulift` can be golfed with `small_map`, but I can't figure out how to do the same with `small_lift`.

Reply:
It's only a few symbols longer (not tested). Without this, your PR makes `small_max` classical.

```suggestion
let ⟨⟨γ, ⟨f⟩⟩⟩ := hα in small.mk' $ f.trans equiv.ulift.symm
```

Otherwise LGTM. Thanks!
bors d+